benchmark/benchmark_device_memory.cpp

```diff
@@ -363,6 +363,10 @@ void run_benchmark_memcpy(benchmark::State& state, size_t size, const hipStream_
     T* d_output;
     HIP_CHECK(hipMalloc(reinterpret_cast<void**>(&d_input), size * sizeof(T)));
     HIP_CHECK(hipMalloc(reinterpret_cast<void**>(&d_output), size * sizeof(T)));
+
+    // Copy input from host to device
+    HIP_CHECK(hipMemcpy(d_input, input.data(), input.size() * sizeof(T), hipMemcpyHostToDevice));
```

Comment:
Good find there...although looking deeper into this code, it makes me question whether we need this hipMemcpy at all, and going even further, do we even need the host input vector and all the data initialization?  We're just measuring raw hipMemcpy speeds, the content of the data buffer shouldn't matter.  We don't check the contents of the output buffer either so it seems like at no point do we care about the data contents.  @Naraenda do you concur?

Reply:
I was wondering the same thing - the data content probably doesn't matter here. I made a similar change for the memcpy benchmark in the rocPRIM static analysis task ([here](https://github.com/ROCmSoftwarePlatform/rocPRIM/pull/470/files#diff-f5aee3bfaaed9760c714f5e805e0421ffedc05dc82d847887ddd02f489843329R441)), so whatever we decide to do here I can also change there.

Reply:
I've updated the code here to remove the data initialization and host to device data transfer.